todocli/task.py

```diff
@@ -0,0 +1,56 @@
+from datetime import datetime
+from enum import Enum
+
+from todocli.todo_api_util import api_timestamp_to_datetime
+
+
+class Task:
+    class Status(Enum):
+        Completed = "completed"
+        NotStarted = "notStarted"
+        InProgress = "inProgress"
+        WaitingOnOthers = "waitingOnOthers"
+        Deferred = "deferred"
```

Comment:
I don't think it's necessary to set these as strings if we're using it as `Enum`. 
```suggestion
        Completed = 0
        NotStarted = 1
        InProgress = 2
        WaitingOnOthers = 3
        Deferred = 4
```

Reply:
I see. In that case, for accessing strings, we could use ENUM.name to access the string part!
`self.request["status"] = status.name`

Reply:
Hmmm, I just checked it and I'm not sure if I understood you correctly...

ENUM.name will print the name of the enum constant, not its value...
 
So:
 ```
val = Task.Status.NotStarted
print(val.name)
``` 
will print "NotStarted". But the API expects "notStarted" (first character lower case).

So we could rename the Enum values to match what the API expects (e.g. to `Task.Status.notStarted`).
This could lead to problems if any API value would for example start with a number, as enum names cannot begin with a number.
Using the ENUM.value with a string will ensure that any possible value can be expressed, that way we are not limited by the python syntax...

While this might be a "non-conventional" use, I don't really see a problem using enums with string values. 
What do you think?

Reply:
Yup. I was suggesting to change the enum name to match API values. But perhaps you do have a point in that there could be issues in the future. Lets leave it as is. Thanks!